Jellyfin.Api/Controllers/LibraryStructureController.cs

```diff
@@ -81,6 +81,12 @@ public async Task<ActionResult> AddVirtualFolder(
         [FromBody] AddVirtualFolderDto? libraryOptionsDto,
         [FromQuery] bool refreshLibrary = false)
     {
+        // Windows does not allow files or folders with names that has leading or trailing spaces
+        if (name.Length != name.Trim().Length)
```

Comment:
This kind of validation should preferably be done with a regex attribute so it is added to the open api spec. Additionally web changes need to be made to implement this same validation locally to hint a user about invalid values.